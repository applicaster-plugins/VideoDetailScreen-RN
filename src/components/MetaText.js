import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import { styles } from './MetaText.styles';

class MetaText extends PureComponent {
  constructor(props) {
    super(props);
    // deferred for layout data to be ready
    this.styles = styles();
  }

  render() {
    const { children } = this.props;
    return (
      <View style={this.styles.container}>
        <Text style={this.styles.text}>{children}</Text>
      </View>
    );
  }
}

MetaText.propTypes = {
  children: PropTypes.string
};

export default MetaText;
