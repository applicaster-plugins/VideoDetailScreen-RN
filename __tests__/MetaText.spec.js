import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import MetaText from '../src/components/MetaText';

jest.mock('../src/utils/zappConfig');

it('MetaText renders correctly', () => {
  const tree = renderer.create(<MetaText>Test Meta Text</MetaText>).toJSON();

  expect(tree).toMatchSnapshot();
});
