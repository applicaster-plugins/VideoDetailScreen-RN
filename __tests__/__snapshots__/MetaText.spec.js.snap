// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaText renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "justifyContent": "flex-start",
      "marginVertical": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(239, 239, 239)",
        "fontFamily": "Exo2-Medium",
        "fontSize": 13,
        "fontStyle": "normal",
        "fontWeight": "500",
        "letterSpacing": 0,
        "lineHeight": 16,
        "textAlign": "left",
      }
    }
  >
    Test Meta Text
  </Text>
</View>
`;
